src/python/pants/backend/python/dependency_inference/module_mapper.py

```diff
@@ -361,7 +382,12 @@ def add_modules(modules: Iterable[str], *, type_stub: bool = False) -> None:
                     )
                 add_modules(stub_modules, type_stub=True)
             else:
-                add_modules(DEFAULT_MODULE_MAPPING.get(proj_name, (fallback_value,)))
+                add_modules(
+                    DEFAULT_MODULE_MAPPING.get(
+                        proj_name,
+                        generate_mappings_from_pattern(proj_name, fallback_value),
+                    )
+                )
```

Comment:
Related to the comment above, it might make sense to move this into a separate function like:

```python
def generate_mapping(prod_name, fallback_value) -> Iterable[str]:
    if from_mapping := DEFAULT_MODULE_MAPPING.get(proj_name):
        return from_mapping

    if from_patterns := generate_mappings_from_pattern(proj_name):
        return from_patterns
    
    return (fallback_value,)
```

And then this becomes

```suggestion
                add_modules(generate_mappings(proj_name, fallback_value))
```

Reply:
did a version of this. good suggestion.